test/Infrastructure/LeanCode.Serialization.Tests/JsonConvertersTests.cs

```diff
@@ -35,6 +27,39 @@ public void Throws_when_attempting_to_deserialize_DateOnly_with_time_part(string
         Assert.Equal(reserialized, JsonSerializer.Serialize(date, LaxSerializerOptions));
     }
 
+    [Theory]
+    [InlineData("\"2021-12-15T07:23:10.115\"", 7, 23, 10, 115, "\"07:23:10.115\"")]
+    public void Correctly_deserializes_TimeOnly_with_date_part_when_using_lax_converter(
+        string serialized, int hour, int minute, int second, int millisecond, string reserialized)
+    {
+        var time = new TimeOnly(hour, minute, second, millisecond);
+
+        Assert.Equal(time, JsonSerializer.Deserialize<TimeOnly>(serialized, LaxSerializerOptions));
+        Assert.Equal(reserialized, JsonSerializer.Serialize(time, LaxSerializerOptions));
+    }
+
+    [Theory]
+    [InlineData("\"2021-12-15T07:23:10.115Z\"", 2021, 12, 15, 7, 23, 10, 115, 0, "\"2021-12-15T07:23:10.115Z\"")]
+    public void Correctly_deserializes_DateTimeOffset_when_using_lax_converter(
+        string serialized, int year, int month, int day, int hour, int minute, int second, int millisecond, int timeSpanOffset, string reserialized)
+    {
+        var date = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, new TimeSpan(timeSpanOffset,0,0));
+
+        Assert.Equal(date, JsonSerializer.Deserialize<DateTimeOffset>(serialized, LaxSerializerOptions));
+        Assert.Equal(reserialized, JsonSerializer.Serialize(date, LaxSerializerOptions));
+    }
+
+    [Theory]
+    [InlineData("\"2021-12-15T07:23:10.115+03:00\"", 2021, 12, 15, 7, 23, 10, 115, 3, "\"2021-12-15T07:23:10.115Z\"")]
```

Comment:
I believe this is wrong, you are losing the offset when serializing again. The reserialized value should be exactly the same as the serialized input.